### PR TITLE
Glimpse: update api and request shape, optimize and refactor

### PR DESCRIPTION
**Pull request by eddieyu1998**
## Type of change

- [x] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- Update Glimpse prebid api endpoint
- Pass gvlid to storage manager for gdpr use cases
- Include consent info in query params for better efficiency and privacy on the handling of user data
- Update the request body shape to optimize the size of the request
- update fpd shape and optimize fpd object
- Add appendQueryParam and optimizeObject util functions
- Remove unnecessary util functions
- Refactor and rename of some util functions for improved simplicity
- Rename variables for clarity
- Update jsdoc comments
- Reformat the files
- Update test spec: add fpd unit test and remove unnecessary tests

---

## contact email of the adapter’s maintainer
- eddie.yu@glimpseprotocol.io

## Related PR on prebid.github.io

- https://github.com/prebid/prebid.github.io/pull/3669

### CODE REVIEW

**Comment by eddieyu1998** [comment]
After some digging, I've noticed that the reason why gulp lint is failing is probably because of the use of `??` operator in our bid adapter. The version of gulp-eslint used by Prebid currently is v4, which looking at its source code doesn't seem to support the `??` operator.

I'm wondering if Prebid can bump the version of gulp-eslint to a newer version to support new features in Javascript?

**Comment by eddieyu1998** [comment]
I've made a change to replace all the `??` to `||` for now

File: modules/glimpseBidAdapter.js [hunk 1]
```diff
@@ -1,230 +1,201 @@
...
-    return {
-      consentString: gdprConsent.consentString || '',
-      vendorData: gdprConsent.vendorData || {},
-      gdprApplies: hasGdprApplies ? gdprConsent.gdprApplies : true,
-    }
+  if (gdprApplies(bidderRequest)) {
+    const consentString = bidderRequest.gdprConsent.consentString;
+    url = appendQueryParam(url, 'reg', 'gdpr');
+    url = appendQueryParam(url, 'cs', consentString);
+  } else if (ccpaApplies(bidderRequest)) {
```
Comment (msm0504) [inline modules/glimpseBidAdapter.js:111]: Is it possible for both GDPR and CCPA to apply? If so, does your endpoint support multiple `reg` values?

**Review by msm0504 — CHANGES_REQUESTED** [review]

File: modules/glimpseBidAdapter.js (hunk 1 above)
Comment (eddieyu1998) [reply modules/glimpseBidAdapter.js:111]: Is it a requirement that we have to support both GDPR and CCPA when they both applies? We think it'd be unlikely that a user would be classified as under both GDPR and CCPA at the same time, and we currently only consider the GDPR consent if that's the case.

Comment (msm0504) [reply modules/glimpseBidAdapter.js:111]: I checked with other reviewers and it's ok to only consider GDPR in that case.

**Review by msm0504 — APPROVED** [review]

File: modules/glimpseBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/glimpseBidAdapter.js:111]: every user in California browsing a website in the EU has both apply. To test this, set your IP to California and go to Guardian.co.uk and see how it considers this.

Comment (patmmccann) [reply modules/glimpseBidAdapter.js:111]: As another datapoint in support of both frequently applying. Some publishers choose for GDPR to always apply, as considered in this document. https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md

**Pull request merged**